File: frontend/src/types/exam.ts
export type Difficulty = 'beginner' | 'intermediate' | 'advanced' | 'expert';

export interface Exam {
  id: string;
  title: string;
  description: string;
  class_level?: number;
  subject?: string;
  difficulty: Difficulty;
  duration_minutes: number;
  max_score: number;
  max_attempts: number;
  is_active: boolean;
  is_published: boolean;
  created_at: string;
}

export interface Question {
  id: string;
  exam_id: string;
  title: string;
  description: string;
  difficulty: Difficulty;
  language: string;
  time_limit_seconds: number;
  memory_limit_mb: number;
  max_score: number;
  question_order: number;
  is_active?: boolean;
}

export interface TestCase {
  id: string;
  question_id: string;
  test_case_order: number;
  input_data: string;
  expected_output: string;
  is_hidden: boolean;
  weight: number;
  timeout_seconds: number;
}

export type AttemptStatus = 'STARTED' | 'IN_PROGRESS' | 'SUBMITTED' | 'EVALUATING' | 'COMPLETED' | 'EXPIRED' | 'CANCELLED';

export interface ExamAttempt {
  id: string;
  candidate_id: string;
  exam_id: string;
  status: AttemptStatus;
  started_at: string;
  expires_at: string;
  remaining_seconds: number;
  submitted_at?: string;
  completed_at?: string;
  total_score: number;
  max_score: number;
}

File: backend/app/schemas/exam.py
from pydantic import BaseModel, Field
from typing import Optional, Any

class ExamCreateSchema(BaseModel):
    title: str
    description: str
    class_level: int = Field(default=10, ge=1, le=12)
    subject: str = Field(default="Mathematics")
    language: str = Field(default="English")
    difficulty: str = "intermediate"
    duration_minutes: int = Field(default=60, gt=0)
    max_score: float = Field(default=100.0, gt=0.0, le=100.0)
    max_attempts: int = Field(default=1, gt=0)

class ExamUpdateSchema(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    class_level: Optional[int] = Field(default=None, ge=1, le=12)
    subject: Optional[str] = None
    language: Optional[str] = None
    difficulty: Optional[str] = None
    duration_minutes: Optional[int] = Field(default=None, gt=0)
    max_score: Optional[float] = Field(default=None, gt=0.0, le=100.0)
    max_attempts: Optional[int] = Field(default=None, gt=0)
    is_active: Optional[bool] = None

class ExamResponseSchema(BaseModel):
    id: str
    title: str
    description: str
    class_level: int = 10
    subject: str = "Mathematics"
    language: str = "English"
    difficulty: str
    duration_minutes: int
    max_score: float
    max_attempts: int
    is_active: bool
    is_published: bool
    created_at: Any
